src/haddock/libs/libalign.py

```diff
@@ -0,0 +1,505 @@
+
+"""library of functions to perform sequence and structural alignments."""
+import os
+import shlex
+import subprocess
+from functools import partial
+from pathlib import Path
+
+import numpy as np
+from Bio import Align
+from Bio.Align import substitution_matrices
+from Bio.Seq import Seq
+
+from haddock import log
+from haddock.libs.libontology import PDBFile
+from haddock.libs.libpdb import split_by_chain
+
+
+RES_TO_BE_IGNORED = ["SHA", "WAT"]
+
+PROT_RES = [
+    "ALA",
+    "ARG",
+    "ASN",
+    "ASP",
+    "CYS",
+    "GLN",
+    "GLU",
+    "GLY",
+    "HIS",
+    "ILE",
+    "LEU",
+    "LYS",
+    "MET",
+    "PHE",
+    "PRO",
+    "SER",
+    "THR",
+    "TRP",
+    "TYR",
+    "VAL",
+    ]
+
+DNA_RES = ["DA", "DC", "DT", "DG"]
+# Backbone
+PROT_ATOMS = ["C", "N", "CA", "O"]
+# Bases
+DNA_ATOMS = [
+    "C5",
+    "N9",
+    "N2",
+    "C8",
+    "O2",
+    "N4",
+    "N7",
+    "C7",
+    "N1",
+    "N6",
+    "C2",
+    "O4",
+    "C6",
+    "N3",
+    "C4",
+    "O6",
+    ]
+
+
+class ALIGNError(Exception):
+    """Raised when something goes wrong with the ALIGNMENT library."""
+
+    def __init__(self, msg=""):
+        self.msg = msg
+        super().__init__(self.msg)
+
+
+def calc_rmsd(V, W):
+    """Calculate the RMSD from two vectors."""
+    diff = np.array(V) - np.array(W)
+    N = len(V)
+    return np.sqrt((diff * diff).sum() / N)
+
+
+def kabsch(P, Q):
+    """Find the rotation matrix using Kabsch algorithm."""
```

Comment:
What is `P` and what is `Q`? Do they need to have any specific dimensions?

To render docs the best way, I would suggest:

```python
"""
Find the rotation matrix using Kabsch algorithm.

`Kabsch algorithm <https://en.wikipedia.org/wiki/Kabsch_algorithm>`_

Parameters
----------------
vec1 : np.array dtype=float, ndims=(3,), or compatible list
vec2 : np.array dtype=float, ndims=(3,), or compatible list

Returns
-----------
vec3 : np.array, dtype=float, ndims=?
    The resulting Kabsch calculation vector.
"""
```

I find it important to specify the input and output dimensions

Reply:
Agreed, I will try to setup a documentation for this library. Remember that this is quite preliminary, so the docs could change substantially in the near future

Reply:
No worries if it is preliminary. Just write for what you have :wink:

Remember the docs should explain how to use the function without seeing the implementation code. For the style we use `numpydoc` style, yes.

You can try in your python prompt:

```python
from haddock.libs.libalign import kabsch


help(kabsch)
```

And see the outcome. Write the docstrings such that the help shows what you want to see :wink:

Great job!